imap/xapian_wrap.cpp

```diff
@@ -1718,6 +1726,33 @@ static Xapian::Query *query_new_email(const xapian_db_t *db,
         q |= db->parser->parse_query(str, qpflags, prefix);
     }
 
+    // query localpart@domain
+    if ((atsign > str) && atsign[1]) {
+        struct address *addr = NULL;
+        char *a = NULL;
+
+        parseaddr_list(str, &addr);
+        if (addr && (a = address_get_all(addr, /*canon_domain*/1))) {
```

Comment:
Per discussions on slack - we need to detect if there's a wildcard in the address, and if so fall back to the 'D' and 'L' keys .  Only exact matches should use the 'A' key.

Reply:
Resolved with the check below